test/test_jobs/__init__.py

```diff
@@ -216,3 +231,148 @@ def test_get_quarterly_date_returns_same_time_as_passed(self):
+    def test_completed_tasks_cleanup_scheduled_with_weekly_queue(self, non_contr, inactive,
+            autoimport, project, export):
+        """Test completed_tasks_cleanup gets scheduled."""
+        inactive.return_value = jobs()
+        non_contr.return_value = jobs()
+
+        project = ProjectFactory.create(info=dict(completed_tasks_cleanup_days=30))
+        weekly_jobs = get_periodic_jobs("weekly")
+        # Only returns jobs for the specified queue
+        for job in weekly_jobs:
+            assert job["name"].__name__ == "perform_completed_tasks_cleanup", "completed tasks cleanup job should be scheduled"
+            assert job['queue'] == "weekly"
+
+    @with_context
+    @patch('pybossa.jobs.purge_task_data')
+    def test_completed_tasks_cleanup(self, mock_purge_tasks):
+        """Test completed_tasks_cleanup deletes tasks qualify for deletion."""
+
+        cleanup_days = 30
+        project = ProjectFactory.create(info=dict(completed_tasks_cleanup_days=cleanup_days))
+
+        # task creation dates. generate sample tasks
+        # 2 tasks completed are with creation date more than 30 days from current date
+        # 1 task completed and is with current creation date
+        now = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S')
+        created = (datetime.utcnow() - timedelta(60)).strftime('%Y-%m-%dT%H:%M:%S')
+        past_30days = (datetime.utcnow() - timedelta(30)).strftime('%Y-%m-%dT%H:%M:%S')
+
+        task1 = TaskFactory.create(project=project, created=past_30days, n_answers=2, state="completed")
+        task2 = TaskFactory.create(project=project, created=now, n_answers=4, state="completed")
+        task3 = TaskFactory.create(project=project, created=past_30days, n_answers=3, state="completed")
+
+        TaskRunFactory.create_batch(2, project=project, created=created, finish_time=now, task=task1)
+        TaskRunFactory.create_batch(4, project=project, created=created, finish_time=now, task=task2)
+        TaskRunFactory.create_batch(3, project=project, created=created, finish_time=now, task=task3)
+
+        task1_id, task2_id, task3_id = task1.id, task2.id, task3.id
+        project_id = project.id
+        perform_completed_tasks_cleanup()
+        assert mock_purge_tasks.call_count == 2
+        # task 1 and task 3 would be cleaned up as they are completed
+        # and 30 days old hence qualifying for deletion.
+        # task 2 though complete is less than 30 days old, hence not
+        # get called for deletion
+        expected_calls_params = [(task3_id, project.id), (task1_id, project.id)]
+        actual_calls_params = []
+        for call in mock_purge_tasks.call_args_list:
+            assert call[0] in expected_calls_params
+            actual_calls_params.append(call[0])
+        assert (task2_id, project_id) not in actual_calls_params, "Task id 2 should not be purged"
+
+        # To test actual task cleanup against database, make sure to have
```

Comment:
Would it make sense to move the commented-out unit test code into a separate commented-out unit test method - just to keep it isolated and not forgotten?  *Minor.*